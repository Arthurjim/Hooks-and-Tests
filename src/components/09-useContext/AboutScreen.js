import React, { useContext } from 'react'
import { UserContext } from './UserContext'

const AboutScreen = () => {
    const {user, setUser} = useContext(UserContext)
    const handleClcik =()=>{
        setUser({})
    }
    return (
        <div>
            <h1>AboutScreen</h1>
            <hr/>
            <pre>
                {JSON.stringify(user, null,3)}
            </pre>
            <button className="bnt btn-secondary" onClick={handleClcik}>
                Logout
            </button>
        </div>
    )
}

export default AboutScreen