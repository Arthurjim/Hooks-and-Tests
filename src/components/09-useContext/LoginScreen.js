import React,{useContext} from 'react'
import { UserContext } from './UserContext'

const LoginScreen = () => {
    const {setUser} = useContext(UserContext)
    const sendUser =()=>{
        const user ={
            id:123,
            name:'Arturo',
            email:'arthurjim013@gmail.com'
        }
        setUser(user)
    }
    return (
        <div>
            <h1>LoginScreen</h1>
            <hr/>
                <button className="btn btn-primary" onClick={sendUser}>
                    Ingresar
                </button>
        </div>
    )
}

export default LoginScreen