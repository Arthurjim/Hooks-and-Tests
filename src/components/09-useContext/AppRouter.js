
import React from "react";
import {
  BrowserRouter as Router,
  Routes ,
  Route,
  Navigate
} from "react-router-dom";
import AboutScreen from './AboutScreen'
import LoginScreen from './LoginScreen'
import HomeScreen from './HomeScreen'
import NavBar from "./components/NavBar";

const AppRouter = () => {
    return (
       <Router>
           <div>
               <NavBar/>
               <div className="container">

                    <Routes >
                        <Route exact path="/"  element={<HomeScreen />}/>
                        <Route exact path="/about" element={<AboutScreen />}/>
                        <Route exact path="/login" element={<LoginScreen />}/>
                        {/* <Route element={<HomeScreen />}/> */}
                        <Route  path="*" element={<Navigate replace to="/" />} />  
                        
                    </Routes >
               </div>
           </div>
       </Router>
    )
}

export default AppRouter
