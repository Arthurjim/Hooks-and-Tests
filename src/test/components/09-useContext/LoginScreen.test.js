import { mount, shallow } from "enzyme";
import LoginScreen from "../../../components/09-useContext/LoginScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("Preubas en <LoginScreen />", () => {
    const setUser = jest.fn();

    const wrapper = mount(
        <UserContext.Provider value={{setUser}}>
            <LoginScreen />
        </UserContext.Provider>
    )


    test("Debe de mostrarse correctamen", () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test("Debe de ejecutar el setUser con el argumento esperado", () => {
        const user = {
            id: 123,
            name: "Arturo",
            email: "arthurjim013@gmail.com",
        };
        wrapper.find("button").prop("onClick")();
        expect(setUser).toHaveBeenCalledWith(user);
    });
});
