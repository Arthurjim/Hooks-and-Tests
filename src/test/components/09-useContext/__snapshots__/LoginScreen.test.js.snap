// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preubas en <LoginScreen /> Debe de mostrarse correctamen 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Ingresar
  </button>
</div>
`;
